src/deploy-appengine.ts

```diff
@@ -22,7 +22,29 @@ import fs from 'fs';
 export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
 export const GCLOUD_METRICS_LABEL = 'github-actions-deploy-appengine';
 
-async function run(): Promise<void> {
+export function setUrlOutput(output: string): string | undefined {
```

Comment:
Just an idea since regex matching can be brittle and for future patterns as we support for gcloud sdk based actions. 

For grabbing URL can we either 
- (havent tested) pass a --json flag that would return the output as structured json
- since we know the project/name, do a native get request and retrieve the object

Reply:
I will test the formatting flag. For GAE, this is better than Cloud Run, because I can match on `target url: [URL]`. I use the regex because when you deploy a non-default service for GAE or a tag service for Cloud Run you will want the https://service-dot-URL or the https://tag--URL respectively. Though for GAE, the URLs are deterministic and I would have to look at the app.yaml for a non-default service name.

Reply:
That makes sense. In general I was brainstorming a way to make gcloud (and other CLI execs) [more stable](https://cloud.google.com/sdk/docs/scripting-gcloud#:~:text=don't%20depend%20on%20the%20raw%20output%20of%20messages%20printed%20to%20standard%20output.) incase there are upstream formatting changes.

Reply:
I think the JSON or yaml formatting would be more stable. I'll add that as a potential update.